engine/exec/exec.go

```diff
@@ -26,7 +26,7 @@ func Run(ctx context.Context, step *spec.Step, output io.Writer) (*runtime.State
 	cmdArgs := step.Entrypoint[1:]
 	cmdArgs = append(cmdArgs, step.Command...)
 
-	cmd := exec.Command(step.Entrypoint[0], cmdArgs...) //nolint:gosec
+	cmd := exec.CommandContext(ctx, step.Entrypoint[0], cmdArgs...) //nolint:gosec
```

Comment:
the context we are using using here, will that cancel after 10 hrs ? AFAIK we set timeout on ctx if it's provided on a step level. Can you double check ?

Reply:
Added a default timeout in case it's not provided by platform.